arm-timeseriesinsights/2017-02-28-preview/Swagger/timeseriesinsights.json

```diff
@@ -0,0 +1,1780 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "TimeSeriesInsightsClient",
+    "description": "Time Series Insights client",
+    "version": "2017-02-28-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.TimeSeriesInsights/environments/{environmentName}": {
+      "put": {
+        "tags": [
+          "Environments"
+        ],
+        "operationId": "Environments_CreateOrUpdate",
+        "x-ms-examples": {
+          "EnvironmentsCreate": { "$ref": "../examples/EnvironmentsCreate.json" }
+        },
+        "description": "Create or update an environment in the specified subscription and resource group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$",
+            "minLength": 1,
+            "maxLength": 90,
+            "description": "Name of the environment"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnvironmentCreateOrUpdateParameters"
+            },
+            "description": "Parameters for creating an environment resource."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The existing environment definition was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentResource"
+            }
+          },
+          "201": {
+            "description": "The environment create request was accepted. Environment provisioning is an asynchroneous operation. You can periodically get your environment definition and monitor progress via the provisioningState property.",
```

Comment:
This follows the ARM spec for creating resources. We follow the same pattern if an update must be performed asynchronously.
Creating or Updating Resources (PUT/PATCH)
The API flow should be to: 
1.	Respond to the initial PUT request with a 201 Created or 200 OK (per normal guidance);
2.	Since provisioning is not complete, the PUT response body MUST contain a provisioningState set to a non-terminal value (e.g. “Accepted”, or “Created”);
3.	Optional: The response headers may include a Azure-AsyncOperation header pointing to an Operation resource (as described below);
4.	Future GETs on the resource that was created should continue to return a 200 Status Code and provisioningState field that is *non-terminal* as long as the provisioning is in progress;
5.	After the provisioning completes, the provisioningState field should transition to one of the terminal states (as described below). 
6.	The provisioningState field should be returned on all future GETs, even after it is complete, until some other operation (e.g. a DELETE or UPDATE) causes it to transition to a non-terminal state.